src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -195,35 +201,55 @@ private RelationalStorage(string invariantName, string connectionString)
         }
 
 
-        private static async Task<Tuple<IEnumerable<TResult>, int>> ExecuteReaderAsync<TResult>(DbCommand command, Func<IDataRecord, int, TResult> selector)
+        private async Task<Tuple<IEnumerable<TResult>, int>> ExecuteReaderAsync<TResult>(DbCommand command, Func<IDataRecord, int, CancellationToken, Task<TResult>> selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
         {
-            using(var reader = await command.ExecuteReaderAsync().ConfigureAwait(continueOnCapturedContext: false))
-            {
-                return await SelectAsync(reader, selector).ConfigureAwait(false);
+            using(var reader = await command.ExecuteReaderAsync(commandBehavior, cancellationToken).ConfigureAwait(continueOnCapturedContext: false))
+            {                                                                
+                if(cancellationToken.CanBeCanceled && this.SupportsCommandCancellation())
+                {
+                    cancellationToken.Register(CommandCancellation, Tuple.Create(reader, command), useSynchronizationContext: false);
```

Comment:
The resulting CancellationTokenRegistration object should be properly handled. That is, it should be disposed so that the registratated function will be deregistered. It looks like CommandCancellation is resilient against multiple calls or calls after the operation is otherwise gone, but a better practice still.